src/samVisualize/graphGeneration/cfnTemplateGraphGenerator.ts

```diff
@@ -0,0 +1,152 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Graph, GraphObject } from './graph'
+import { yamlParse } from 'yaml-cfn'
+import { LinkTypes } from '../samVisualizeTypes'
+import * as _ from 'lodash'
+import { getLogger } from '../../shared/logger/logger'
+
+/**
+ * @param inputYaml A string representing a YAML template
+ * @returns A JavaScript object corresponding to the given YAML string, or undefined if the input is not valid YAML
+ */
+function yamlStringToObject(inputYaml: string): Record<string, any> | undefined {
+    try {
+        return yamlParse(inputYaml)
+    } catch (err) {
+        // The number of characters previewed in an erroneous template string
+        const characterLimit = 25
```

Comment:
There's a decent amount of discussion about this but here's my 2 cents: you're not using its caller (`generateGraphFromYaml`) outside of tests for now, so I can't tell exactly how you're using this, but if you're not going to use the `inputYaml` string anywhere outside of this file, you should instead pass in the file name and parse the file into a string here (and log that in the error). The first 25 characters of the YAML don't provide a lot of value to someone who's running into this error and isn't sure what's actually being evaluated.

Reply:
So the `generateGraphFromYaml` function will indeed be used outside of tests. When a user executes the command to render a template, the contents of the active editor (open to a CFN template) will be passed into `generateGraphFromYaml`. So there isn't any file reading happening, as the template contents are fetched through the active editor. It works just as the step function rendering currently does. There are error popups from VSCode when the active editor does not contain a valid CFN template. I think it will be clear to the user that the template they just executed the render command on is being evaluated. I was thinking the truncation was just for logging purposes.

Reply:
If the YAML is only being parsed in the context of an editor then truncation makes sense. But if there's other ways to feed in input, then it'd be better to let the error bubble up and let the caller handle it.

Reply:
With the current design, there are no requirements to feed input into `generateGraphFromYaml` other than through the editor. If that changes, I'll change error handling accordingly